### ISSUE TITLE
Conditional Named Group Display w/ "Right to Left" flag

### ISSUE THREAD

**Issue by CarlitoGil** [firasdib/Regex101#1900]
## Bug Description
The pattern is not highlighted properly when a named group is correctly defined after the reference in a right-to-left regex.

### Reproduction steps
1. Choose .NET flavor
2. Add "R" flag (right-to-left)
3. Use pattern `(?(numbers)[A-Z]+|\W+)(?<numbers>\d+)?`
4. Use test string `ABC123XYZ`

### Expected Outcome
`(?(numbers)` should be highlighted and tooltip'd as a "conditional" that depends on the value of the named group "numbers".
Currently, `(numbers)`  is displayed incorrectly as a positive look-ahead.

### Notes
The engine correctly matches `ABC123` in the above example.
For comparison: without the "R" flag, use pattern `(?<numbers>\d+)?(?(numbers)[A-Z]+|\W+)`

**Comment by firasdib** [comment]
RTL seems to adjust the parsing of the regex as well, not just how it is applied to the string, which is interesting. I wonder what other edge cases there are to consider...

**Comment by firasdib** [comment]
Perhaps this is a discrepancy of .NET 7?

The site uses .NET 7, and without RTL, your pattern doesn't match https://regex101.com/r/rUFyGK/1

**Comment by CarlitoGil** [comment]
> Perhaps this is a discrepancy of .NET 7?

It's not.
In my tests, the engine itself is working as expected, and the same way, in both Regex101 and C# .NET 6.0.
What is failing is simply the display (colored highlight and tooltip) of the regex pattern in Regex101.


> The site uses .NET 7, and without RTL, your pattern doesn't match https://regex101.com/r/rUFyGK/1

This link is not using the test string provided: `ABC<numbers>XYZ`, with angle brackets and literally the string "numbers" to see if the conditional works as a look-ahead (or not) depending on the existence of the named group "numbers" to the left, or right, or not at all.
The non-word characters allow the "else" part of the conditional to match `\W+`.
This test demonstrates that regardless of the state of the RTL flag, with the named group declared **after** the conditional, the engine does not consider `(numbers)` to be a look-ahead, but the pattern is incorrectly highlighted as such.

The problem seems to be that when Regex101 parses the pattern it only accounts for named groups that have been declared before (to the left of) the conditional that references them.

When parsed, `(?(numbers)then|else)` has to be determined to be a look-ahead or a reference to a named group that exists **anywhere in the pattern**, and not just to the right of it, in the same way as `\k<numbers>` is parsed to an error if the group "numbers" does not exist before or after it, independently of the "R" flag.

To put it another way, the tooptip in this image is wrong and does not describe the behavior of the regex engine, be it .NET 6.0, NET 7.0, with or without Right-To-Left.
![image](https://user-images.githubusercontent.com/5706999/190914122-49c1cb75-9b43-40ae-8a10-5f6b622bc0c8.png)
It cannot be a "lookahead" because the named group "numbers" exists in the regex pattern.

**Comment by firasdib** [comment]
Ok! I have misunderstood how the engine parses the regex, thank you for the clarification.

In summary: the condition of a conditional is only considered a positive lookahead if there is no named group that matches anywhere in the regex. Correct?

**Comment by CarlitoGil** [comment]
> In summary: the condition of a conditional is only considered a positive lookahead if there is no named group that matches anywhere in the regex. Correct?

Correct.

When the named group exists, the positive look-ahead has to be explicit, like this: `(?(?=numbers)then|else)`.

Thanks!

**Comment by firasdib** [comment]
This will be working in the upcoming release. Thanks for your patience with me!
